src/main/java/seedu/address/logic/parser/SearchPatientCommandParser.java

```diff
@@ -0,0 +1,53 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.*;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.SearchPatientCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+
+/**
+ * Parses input arguments and creates a new AddPatientCommand object
+ */
+public class SearchPatientCommandParser implements Parser<SearchPatientCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddPatientCommand
+     * and returns an AddPatientCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SearchPatientCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME,  PREFIX_TEMP_RANGE);
+
+        if ((!arePrefixesPresent(argMultimap, PREFIX_NAME)
+                && !arePrefixesPresent(argMultimap, PREFIX_TEMP_RANGE))
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchPatientCommand.MESSAGE_USAGE));
+        }
+
+
```

Comment:
not sure if this will fail the checkstyle because of multiple empty lines

Reply:
will focus on the style!